commands/preupgrade/__init__.py

```diff
@@ -23,6 +23,8 @@
              help='Set preferred channel for the IPU target.',
              choices=['ga', 'tuv', 'e4s', 'eus', 'aus'],
              value_type=str.lower)  # This allows the choices to be case insensitive
+# NOTE(ivasilev) choices argument here will depend on flavor, so not specifying defaults
+@command_opt('target', help='Specify RHEL version to upgrade to')
```

Comment:
@fernflower  I'd like to add choices here to be able to write what are the allowed parameters

Reply:
@fernflower ah...now I got an idea why we do not want that. e.g. because of openstack, tests, .... because in such a case, they should be able to upgrade even to an unsupported systems... maybe we should do a short brainstorming about that.

Reply:
Actually if you are referring to my comment - my comment is a leftover of --flavor parameter, as argparse doesn't allow choices of argX to be dependent on value of argY. 
It is possible to add choices argument now if we want this.

Reply:
I've been thinking about specifying the list of possible values jsut by a function, but I am afraid that people could be confused that on some systems we could print e.g. `[8.4, 8.6]` and on some systems `[8.2]` because of different detected flavours.

Reply:
@Rezney @fernflower  @vinzenz @shaded-enmity : btw, still not sure about printing the choices. I mean from the point that based on detected flavours, people will see different choices. or maybe we could have even "dynamic" help. I mean. description detected flavour and than the choices.. something like:
```
Specify the target version of RHEL system. See URL  for the list of supported upgrade paths. For your system, we detected {} "flavour" (different flavours could mean different possible choices where to upgrade).
```
Take that just as an idea - definitely this wording is not good. E.g. how to explain what the flavour is..

Reply:
Despite feeling that the dynamic approach might not be exactly "kosher" in the context of Linux command help, this way (adding dynamic choices), we would be telling users what target system versions they can upgrade to without sending them to read documentation straight away.

What about: "Specify RHEL version to upgrade to. For this system (SAP flavour), Leapp detected the following available paths: ... ..." ?

Reply:
I think of adding this to choices and generating a dynamic "Here are available choices for detected {} flavor" message. This way argparse will take care of vetting the arguments and warning the user for us. wdyt?